axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java

```diff
@@ -188,68 +187,24 @@ public EventDispatcher(EventStoreLocator eventStoreLocator,
-                                                    t -> {
-                                                        sink.error(t);
-                                                        removeTrackerInfo(trackerInfo);
-                                                    },
-                                                    () -> {
-                                                        sink.complete();
-                                                        removeTrackerInfo(trackerInfo);
-                                                    });
-
-                        trackingEventProcessors.computeIfAbsent(new ClientStreamIdentification(trackerInfo.context,
-                                                                                               trackerInfo.client),
-                                                                key -> new CopyOnWriteArrayList<>()).add(trackerInfo);
-                        logger.info("Starting tracking event processor for {}:{} - {}",
-                                    request.getClientId(),
-                                    request.getComponentName(),
-                                    request.getTrackingToken());
-                    }
-                    Sinks.Many<GetEventsRequest> eventStoreRequestFlux = eventStoreRequestFluxRef.get();
-                    if (eventStoreRequestFlux != null) {
-                        if (eventStoreRequestFlux.tryEmitNext(request).isFailure()) {
-                            eventStoreRequestFlux.tryEmitError(new RuntimeException("Unable to send the request for events."));
-                            removeTrackerInfo(eventTrackerInfoRef.get());
-                        }
-                    }
-                }, t -> {
-                    Sinks.Many<GetEventsRequest> eventStoreRequestFlux = eventStoreRequestFluxRef.get();
-                    if (eventStoreRequestFlux != null) {
-                        eventStoreRequestFlux.tryEmitError(t);
-                    }
-                    removeTrackerInfo(eventTrackerInfoRef.get());
-                }, () -> {
-                    Sinks.Many<GetEventsRequest> eventStoreRequestFlux = eventStoreRequestFluxRef.get();
-                    if (eventStoreRequestFlux != null) {
-                        eventStoreRequestFlux.tryEmitComplete();
-                    }
-                    removeTrackerInfo(eventTrackerInfoRef.get());
-                }));
+        return requestFlux.switchOnFirst((signal, rf) -> {
+            if (signal.isOnNext()) {
+                GetEventsRequest request = signal.get();
+                EventTrackerInfo trackerInfo = new EventTrackerInfo(request.getClientId(),
+                                                                    context,
+                                                                    request.getTrackingToken() - 1);
+                return eventStoreLocator.eventStore(context, request.getForceReadFromLeader())
```

Comment:
add trackerInfo to trackingEventProcessors (see original line 225)

Reply:
This is correct! Nice catch!